bigquery_etl/glam/probe_counts.py

```diff
@@ -0,0 +1,108 @@
+r"""Metric counting.
+
+```bash
+python3 -m bigquery_etl.glam.probe_counts
+
+diff \
+    <(cat sql/telemetry_derived/clients_histogram_probe_counts_v1/query.sql) \
+    <(python3 -m bigquery_etl.glam.probe_counts)
+```
+"""
+from argparse import ArgumentParser
+from itertools import combinations
+from typing import List
+
+from jinja2 import Environment, PackageLoader
+
+from bigquery_etl.format_sql.formatter import reformat
+
+
+def render_query(attributes: List[str], **kwargs) -> str:
+    """Render the main query."""
+    env = Environment(loader=PackageLoader("bigquery_etl", "glam/templates"))
+    sql = env.get_template("probe_counts_v1.sql")
+
+    # only compute a shallow set of attributes to avoid excess complexity
+    max_combinations = 3
+    attribute_combinations = []
+    for subset_size in reversed(range(max_combinations + 1)):
+        for grouping in combinations(attributes, subset_size):
+            select_expr = []
+            for attribute in attributes:
+                select_expr.append((attribute, attribute in grouping))
+            attribute_combinations.append(select_expr)
```

Comment:
In the Glam front-end, both `channel` and `app_version` must be specified, so it's not necessary to have the combinations where these are `NULL`, especially if it can speed things up. The only reason to have them in the dataset is if someone wants to look at them outside of the Glam front-end. The desktop telemetry glam ETL doesn't need these either.

Reply:
I was curious about this, I noticed that I generated more lines than the original, but I couldn't figure out the pattern.